src/parsing/trigger_parsing.cpp

```diff
@@ -637,9 +655,6 @@ int32_t simplify_trigger(uint16_t *source) {
 				source_size -= 2;
 			}
 		}
-
-		// MISSING OPTIMIZATION: fold inner 'and' scope into outer 'and' scope, inner 'or' scope into outer 'or' scope
```

Comment:
Instead it should run here. This is because it needs the removal of extraneous negation to run properly